src/main/java/com/flansmod/common/guns/ItemGun.java

```diff
@@ -458,58 +496,38 @@ private void onUpdateClient(ItemStack itemstack, World world, Entity entity, int
-								isNightVision = true;
-							if(type.allowSlow)
-								isSlow = true;*/
-                            FlansModClient.currentScope = currentScope;
-                            FlansModClient.lastZoomLevel = currentScope.hasVariableZoom()?getCurrentVariableZoom(itemstack):currentScope.getZoomFactor();
-                            FlansModClient.lastFOVZoomLevel = currentScope.hasVariableZoom()?1F:currentScope.getFOVFactor();
-                            float f = FlansModClient.originalMouseSensitivity = gameSettings.mouseSensitivity;
-                            gameSettings.mouseSensitivity = f / (float) Math.sqrt(currentScope.hasVariableZoom()?getCurrentVariableZoom(itemstack):currentScope.getZoomFactor());
-                            FlansModClient.originalThirdPerson = gameSettings.thirdPersonView;
-                            gameSettings.thirdPersonView = 0;
-
-                            // This fixes some of the issues with FOV getting changed by repeated scoping and unscoping.
-                            // In effect, it says that we should only set the 'original' zoom of the client if we know the FOV is
-                            // unaffected from scoping.
-                            if (Math.abs(FlansModClient.zoomProgress - FlansModClient.lastZoomProgress) < 0.0001F) {
-                                FlansModClient.originalFOV = gameSettings.fovSetting;
-                            }
-
-                            //Send ads spread packet to server
-                            sendADSSpreadToServer(itemstack, player.isSneaking(), player.isSprinting());
-                            isScoped = FlansModClient.currentScope != null;
-                            FlansMod.getPacketHandler().sendToServer(new PacketGunState(isScoped));
-                        }
-                        FlansModClient.scopeTime = 10;
-                    } else {
-                        if (!Mouse.isButtonDown(FlansModClient.aimButton.getButton())) {
-                            if (FlansModClient.currentScope != null) {
-                                FlansModClient.currentScope = null;
-                                gameSettings.mouseSensitivity = FlansModClient.originalMouseSensitivity;
-                                gameSettings.thirdPersonView = FlansModClient.originalThirdPerson;
-                                gameSettings.fovSetting = FlansModClient.originalFOV;
-
-                                //Send default spread packet to server
-                                FlansMod.getPacketHandler().sendToServer(new PacketGunSpread(itemstack, type.getDefaultSpread(itemstack)));
-                                isScoped = FlansModClient.currentScope != null;
-                                FlansMod.getPacketHandler().sendToServer(new PacketGunState(isScoped));
-                            }
-                        }
                     }
                 }
             }
         }
+
         if (soundDelay > 0) {
```

Comment:
I might be wrong, but does the return statements for item not being held in hand stop this sound delay from ticking down?

It's a bit hard to see in github so I might be wrong

Reply:
Yes it would, could set it to 0 when its determined to no longer be in hand, what do you think?

Reply:
I mostly made this change to remove some excessive nesting

Reply:
Could just put the if () { sound delay-- }
Above the inhand check perhaps? So previous behaviour stays the same

Reply:
I believe that would cause it to play the sound earlier since it would be decrementing the value before it checks if it should play

Reply:
Hmm.. decrement in the return condition, and in the end too?

Reply:
Will do that for now, should maybe be revisited later on